src/Encoding.chpl

```diff
@@ -0,0 +1,254 @@
+/* Documentation for Encoding */
+module Encoding {
+
+  use IO;
+
+  // wireTypes
+  const varint = 0;
+  const lengthDelimited = 2;
+  const fixed64Type = 1;
+  const fixed32Type = 5;
+
+  proc unsignedVarintAppend(val:uint, ch: channel(true,iokind.little,false)) throws {
+    if val == 0 {
+      ch.write(b"\x00");
```

Comment:
It is not a big deal but I think this is better written
```
var zero: uint(8);
ch.write(zero);
```

I understand that is longer (it's just that it keeps the code using simpler I/O).

Reply:
Yeah, I will make the change :)